src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -32,26 +32,53 @@ public class DefaultOperations {
 	static {
 		// Number - Number
 		Arithmetics.registerOperation(Operator.ADDITION, Number.class, (left, right) -> {
-			if (Utils.isInteger(left, right))
-				return left.longValue() + right.longValue();
+			if (Utils.isInteger(left, right)) {
+				long result = left.longValue() + right.longValue();
+				// catches overflow, from Math.addExact(long, long)
+				if (((left.longValue() ^ result) & (right.longValue() ^ result)) >= 0)
```

Comment:
```suggestion
				if ((left.longValue() ^ result) & (right.longValue() ^ result) >= 0)
```

Reply:
no this is actually useful imo
`(x ^ y) & (a ^ b) >= 0` can be interpreted as `(x ^ y) & ((a ^ b) >= 0)`
technically it's not necessary in java, since java doesn't treat booleans as integers (idk why), but I still think it's good to have for clarity